dynet/nodes-dropout.cc

```diff
@@ -30,7 +30,10 @@ size_t Dropout::aux_storage_size() const {
 template<class MyDevice>
 void Dropout::forward_dev_impl(const MyDevice & dev, const vector<const Tensor*>& xs, Tensor& fx) const {
   Tensor m(dim, (float*)aux_mem, fx.device, DeviceMempool::FXS);
-  TensorTools::randomize_bernoulli(m, (1.f-p), 1.f / (1.f-p));
+  std::uniform_int_distribution<> seed_dist(1, 2147483647);
```

Comment:
I'm not a huge fan of this code being copied all the time. If this is something that we'll want to use frequently, perhaps it can be made a global variable in the DyNet namespace or something?

Also, tangentially, there are some random number generation functions in `tensor.h`/`tensor.cc` that it might be reasonable to move to their own `rand.h`/`rand.cc` files, along with this one. What do you think?

Reply:
That sounds good, I moved the random helpers to their own files and added a draw_random_seed() helper to avoid code duplication.